### PR TITLE
examples: minter demo addition

### PR DESCRIPTION
**Pull request by hachreak**
Signed-off-by: Leonardo Rossi leonardo.r@cern.ch

### CODE REVIEW

**Comment by jirikuncar** [comment]
- `examples/data/marc21/bibliographic.xml` should not be in this package. Use the one from `invenio-records`.
- IMHO you can merge the example apps into one.

**Comment by hachreak** [comment]
it's a demo. I think make sense to make it self-contained.

**Comment by hachreak** [comment]
ping @jirikuncar

File: examples/app.py [hunk 1]
```diff
@@ -22,15 +22,38 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
+r"""Minimal Flask application example for development.
 
-"""Minimal Flask application example for development.
+Install `uuid` package:
```
Comment (jirikuncar) [inline examples/app.py]: you don't need it

File: examples/app.py [hunk 2]
```diff
@@ -22,15 +22,38 @@
...
+invenio_records/data/marc21/bibliographic.xml):
 
 .. code-block:: console
 
-   $ cd examples
-   $ python app.py
+    $ demomarc21nbrecs=$(grep -c '</record>' data/marc21/bibliographic.xml)
+    $ dojson do -i data/marc21/bibliographic.xml -l marcxml marc21 | \
+        flask -a app.py records create \
+        $(for i in $(seq 1 $demomarc21nbrecs); do echo "-i " $(uuid); done)
```
Comment (jirikuncar) [inline examples/app.py]: not needed

File: examples/app.py [hunk 3]
```diff
@@ -84,3 +113,11 @@ def oaiserver():
     }
     Record.create({'title': 'Test0', '$schema': schema})
     db.session.commit()
+
+    # mint all records
+    records = RecordMetadata.query.all()
+    for record in records:
+        recid_minter(record.id, record.json)
+        oaiid_minter(record.id, record.json)
+        # mark record as modified
```
Comment (jirikuncar) [inline examples/app.py]: https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L170 !!!

File: invenio_oaiserver/response.py [hunk 4]
```diff
@@ -224,14 +225,18 @@ def header(parent, identifier, datestamp, sets=None, deleted=False):
...
     """Create OAI-PMH response for verb Identify."""
+    record_dumper = etree_dumper(**kwargs)
     e_tree, e_getrecord = verb(**kwargs)
 
-    record = RecordMetadata.query.get(kwargs['identifier'])
+    pid = OAIIDProvider.get(pid_value=kwargs['identifier']).pid
+    record = RecordMetadata.query.get(pid.object_uuid)
     header(
         e_getrecord,
         identifier=str(record.id),
```
Comment (jirikuncar) [inline invenio_oaiserver/response.py]: `identifier=str(pid.object_uuid)`

File: tests/test_verbs.py [hunk 5]
```diff
@@ -58,6 +66,54 @@ def test_identify(app):
...
+
+        with app.test_client() as c:
+            result = c.get(
+                "/oai2d?verb=GetRecord&identifier={0}&metadataPrefix=oai_dc"
+                .format(pid_value))
+            assert 200 == result.status_code
+
+            tree = etree.fromstring(result.data)
+
+            namespaces = {'x': NS_OAIPMH}
```
Comment (jirikuncar) [inline tests/test_verbs.py:96]: `invenio_oaiserver.response:NSMAP` -> no need for `x:`

Comment (hachreak) [inline tests/test_verbs.py:96]: what do you mean?

Comment (jirikuncar) [inline tests/test_verbs.py:96]: `NSMAP = {None: '...'}` -> `tree.xpath('/OAI-PHM', namespaces=NSMAP)`

**Pull request merged**